### ISSUE TITLE
Analytics Package Setup Redux

### ISSUE THREAD

**Issue by meneal** [CSC510-2015-Axitron/maze#90]
This needs to be done again once the like/dislike box is functional.

**Comment by meneal** [comment]
This is in there now.  I don't really like how the data is coming out a whole lot in keen, but I think this pretty much gets the job done.  This is what the objects look like, and they can be queried in the dashboard for keen.io.

`{`
    `"item": "Procedurally Generated",`
    `"liked": "No",`
    `"keen": {`
        `"timestamp": "2015-04-02T22:42:13.794Z",`
        `"created_at": "2015-04-02T22:42:13.794Z",`
       ` "id": "551dc5c5672e6c493bb4d188"`
   ` }`
`}   `


Basically item changes from handwritten to procedurally generated and liked changes from no to yes.  Let me know if you have an opinion on how I'm encoding the data or if it should be changed.  Additionally we could add some sort of static page the shows the actual output graphs.  Let me know if anyone thinks that is necessary beyond what's being shown in the explorer in keen.

**Comment by duh102** [comment]
How "not like" exactly, is it not looking nice when viewed in a graph?

**Comment by meneal** [comment]
Basically within the keen.io explorer you can't do a side by side comparison.  So I can't figure out a way to show anything other than something like the number of people who liked procedural levels, and the number of people who liked handwritten levels, or the inverse of that.  So you can't really show them together in the explorer.  

I honestly can't think of a way that would make the data come out nicer in a graph that I can generate with the explorer.

**Comment by duh102** [comment]
Well, I looked it up and there's a way to do multiple group-by's, which is what we want, but the graph that keen.io then produces doesn't make a sensible graph, it labels the pie segments by the first group only instead of by both. If we had our own graph-making (or went with a table) we could use the JSON query version of keen.io and not worry about it, but I'm uncertain how difficult making graphs using a third party lib is. For reference, the URL I made is https://api.keen.io/3.0/projects/551ad76c59949a707edddefa/queries/count?api_key=(apikey)&event_collection=sentiment&analysis_type=count&group_by=%5B%22item%22%2C%20%22liked%22%5D&timezone=UTC

**Comment by meneal** [comment]
It's actually possible to use [this](https://github.com/keen/keen-js) to generate custom graphs, but I honestly just don't have enough time to really work with it enough to get it working.  I started working on it a bit, but just realized that it would be really time consuming.

I guess the thing is that we're collecting the data, and we can do some really simple graphs through keen.  Do you think that's enough?

**Comment by duh102** [comment]
So it seems there's a weird problem with keen.io that I don't know what's wrong with, the query part of it doesn't seem to work with Firefox? And we'd kinda need that no matter what sort of thing we do, which is annoying me. Have you encountered this before?

**Comment by duh102** [comment]
Nevermind, figured out why, I had Ghostery enabled and that blocks KeenIO. I should be able to get a simple pie chart page together using it.

**Comment by meneal** [comment]
Which query part are you talking about?  The explorer is working in Firefox.  I just tested it.  The workbench from what I understand is pretty much deprecated.

**Comment by meneal** [comment]
That would be pretty awesome.  I'm working on the python stuff now for
HW3.  Hopefully I'll have a good idea of what's going on with it at our
meeting tomorrow.

**Comment by duh102** [comment]
Just pushed the chart changes, you can view it at /pie.html
They're pretty basic, but I don't think we need to do anything particularly clever with it.

**Comment by meneal** [comment]
That's awesome man!  Definitely makes it something that we can at least
show without logging into the dash during our presentation.  I think we can
definitely close at this point.
